docker-compose.yml

```diff
@@ -0,0 +1,10 @@
+version: '3'
+
+services:
+  app:
+    build: .
+    volumes:
+      - $HOME/.gradle:/root/.gradle
+      - .:/app
```

Comment:
In addition to mounting the host's Gradle cache, this also mounts the host's source repository in the container, which helps power the auto-restart behavior (so local IntelliJ edits are reflected in the container)